pylint/checkers/variables.py

```diff
@@ -1042,6 +1071,7 @@ def _undefined_and_used_before_checker(
             if action is VariableVisitConsumerAction.CONTINUE:
                 continue
             if action is VariableVisitConsumerAction.CONSUME:
+                found_nodes += current_consumer.consumed_uncertain[node.name]
```

Comment:
Can't this be done in `_check_consumer`?

Reply:
There are so many early returns in `_check_consumer` that I would need to write test cases for each of the branches to see if they all need this statement. I was reluctant to possibly introduce subtle bugs. If you have an idea, please let me know!

Reply:
Ah I misunderstood the changes on L1168-1170. I think this might indeed be the best approach. However, perhaps only add a comment to this.
I know I'm asking for a lot of documentation and comments but it is so helpful to be easily understand the code when you're trying to fix a false positive here. As you said, these methods rely heavily on early returns etc. so it can be hard to figure out what is going on.

Reply:
No problem, I agree that some more comments are the way to go here.